apps/train_loc/lib/train_loc/encoder/vehicle_positions_enhanced.ex

```diff
@@ -7,16 +7,47 @@ defmodule TrainLoc.Encoder.VehiclePositionsEnhanced do
 
   alias TrainLoc.Vehicles.Vehicle
 
+  @type feed() :: %{
+          header: feed_header(),
+          entity: [entity()]
+        }
+
+  @type feed_header() :: %{
+          gtfs_realtime_version: String.t(),
+          incrementality: integer(),
+          timestamp: integer()
+        }
+
+  @type entity() :: %{
+          id: non_neg_integer(),
+          vehicle: %{
+            trip:
+              %{start_date: String.t()} | %{start_date: String.t(), trip_short_name: String.t()},
+            vehicle:
+              %{id: non_neg_integer()} | %{id: non_neg_integer(), assignment_status: String.t()},
+            position: %{
+              latitude: float | nil,
+              longitude: float | nil,
+              bearing: 0..359,
+              speed: float()
+            },
+            timestamp: integer()
```

Comment:
nitpick: technically, I think this is also a `non_neg_integer` as all times are after the POSIX epoch.

```suggestion
            timestamp: non_neg_integer()
```

Reply:
This is the result of `DateTime.to_unix/2`: https://github.com/elixir-lang/elixir/blob/f73e318bce04dfc829162e569b2b993df6f2e5df/lib/elixir/lib/calendar/datetime.ex#L778. I'm not sure whether it could actually be negative in practice, but the underlying functions do not appear to guarantee that it will be positive (I could be wrong).

Reply:
Fair enough.